### PR TITLE
[15 min fix] Fixed app code that was causing flaky E2E test

### LINKED ISSUE #13672: Reenable comment on article should pre-populate a comment field when editing
**Describe the bug**

We skipped a test due to unreliable results in  https://github.com/forem/forem/pull/13671

Isolate the issue (it was repeatable for me, inconsistently, locally, so may be a timing related problem) and reenable the test.

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Upon inspection, the E2E test is not flaky and in fact caught some brittle code we had in the front-end code base.

The code changes fix the issue and also give us a little performance boost. The way the code was previously, we registered n events for n comments on an article. Now there is only one event on the containing element of the comments section. With a few comments, not a big deal, but if there are say 50 comments, which I've definitely seen on posts, 1 event being registered instead of 50 is definitely a good thing.

## Related Tickets & Documents

Closes #13672 

## QA Instructions, Screenshots, Recordings

1. Navigate to a post created by you or from seed data on your local.
2. Comment on the post.
3. The new comment will be the top most comment in the Discussion section.
4. Click on the the three dot button at the top right of the newly created comment.
5. The menu should open.

### UI accessibility concerns?

N/A

## Added tests?

- [ ] Yes
- [x] No, and this is why: The reenabled once thought was flaky test covers this.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

Someone saying "Mystery solved!"

### CODE REVIEW

File: cypress/integration/articleFlows/commentOnArticle.spec.js
Comment (djuber) [inline cypress/integration/articleFlows/commentOnArticle.spec.js:369]: :+1: that was fast.

File: app/assets/javascripts/initializers/initializeCommentDropdown.js
Comment (djuber) [inline app/assets/javascripts/initializers/initializeCommentDropdown.js:165]: this makes sense (textually, at least, to a javascript ignoramus).

**Review by djuber — APPROVED** [review]

**Comment by djuber** [comment]
pretty sure this is blocked by the bundle-audit checks (in https://github.com/forem/forem/pull/13675) and a rebase/merge will be needed

File: app/assets/javascripts/initializers/initializeCommentDropdown.js
Comment (nickytonline) [inline app/assets/javascripts/initializers/initializeCommentDropdown.js:88]: This function has been renamed, but the body of it remains the same. The only difference is I pass in an argument now which is used to return the click handler.

**Comment by nickytonline** [comment]
I'll rebase this in the morning once #13675 is in when I start my day, but feel free to rebase, whoever is in before me and merge.

**Review by citizen428 — APPROVED** [review]

**Review by aitchiss — APPROVED** [review]
Great that Cypress is helping us catch these things 👍 

I tested locally on android as well as desktop - all looking good

**Comment by Link2Twenty** [comment]
I'd say `getButton` should be spun out as it's own function that takes `potentialButton` and `commentsContainer` with it's own jsdoc, it would make the code easier to understand but maybe I'm just bias against IIFE 😅.

**Pull request merged**
